kube-runtime/src/cache/mod.rs

```diff
@@ -1,14 +1,18 @@
 //! Caches objects in memory
 
 mod object_ref;
-pub mod store;
+mod store;
 
 pub use self::object_ref::ObjectRef;
-use crate::watcher;
-use futures::{Stream, TryStreamExt};
-use kube_client::Resource;
-use std::hash::Hash;
-pub use store::Store;
+use crate::{utils, watcher};
+use futures::{future::BoxFuture, stream::BoxStream, FutureExt, Stream, StreamExt, TryStreamExt};
+use kube_client::{
+    api::{Api, ListParams},
+    Resource,
+};
+use serde::de::DeserializeOwned;
+use std::{fmt::Debug, hash::Hash};
+pub use store::{Store, Writer};
```

Comment:
The name here is `store::Writer`, not `Writer`. If we absolutely want to hide `store` then it should probably be reexported as `StoreWriter` instead.